Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -39,12 +43,19 @@ void DataReaderTask::init(InitContext& ic)
   ic.services().get<CallbackService>().set(CallbackService::Id::Stop, finishFunction);
 }
 
-void DataReaderTask::sendData(ProcessingContext& pc)
+void DataReaderTask::sendData(ProcessingContext& pc, bool blankframe)
 {
   // mReader.getParsedObjects(mTracklets,mDigits,mTriggers);
-  mReader.getParsedObjects(mTracklets, mDigits, mTriggers);
-
-  LOG(info) << "Sending data onwards with " << mDigits.size() << " Digits and " << mTracklets.size() << " Tracklets and " << mTriggers.size() << " Triggers";
+  if (!blankframe) {
+    mReader.getParsedObjectsandClear(mTracklets, mDigits, mTriggers);
+  } else {
+    //ensure the objects we are sending back are indeed blank.
+    if (mTracklets.size() != 0)
+      LOG(warn) << "Sending data and tracklets should be zero for an empty frame but we have " << mTracklets.size() << " tracklets";
+    if (mTriggers.size() != 0)
+      LOG(warn) << "Sending data and trigger should be zero for an empty frame but we have " << mTriggers.size() << " tracklets";
```

Comment:
How they can be empty if you never clean them? 
Also, I did not understand, do you want to postpone getting rid of intermediate vectors to other PR?

Reply:
I got stuck with documents for cape town, finalising this now.

Reply:
There are more optimisations to come but various excessive copies are removed.